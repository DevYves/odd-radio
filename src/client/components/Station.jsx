import React, { Component } from 'react';
import { onStationSelect, onInfoSelect } from '../util/ClientFunctions.jsx';

class Station extends Component {

  constructor(props) {
    super(props);
    this.state = {
      id: this.props.id,
      name: this.props.name,
      stream: this.props.audioFeed,
      type: this.props.type
    }
    this.onStationSelect = onStationSelect.bind(this);
    this.onInfoSelect = onInfoSelect.bind(this);
  }

  render() {
      console.log('Rendering <Station>')
    return (

      <div>
        <div className={ `container station-container${this.props.stationType}` }>
          <div className="row station-row border">
            <div className="one-third column station-name center"> {this.props.name} </div>
            <div className="one-third column station-play-button center">
              <i className="fa fa-play card-play-button " aria-hidden="true" onClick={ this.onStationSelect }></i>
            </div>
            <div className="one-third column station-info-button center">
              <i className="fa fa-chevron-down card-chevron" aria-hidden="true" onClick={ this.onInfoSelect }></i>
            </div>
          </div>
        </div>
        <div className="container info-container hide-class" id={ this.props.name }>
          <div className="row center-align">
            <div className="station-info center">
              <div className="station-branding one-third column"> { this.props.name } </div>
              <div className="two-thirds column info">
                <div><b>Location:</b> { this.props.city } </div>
                <div><b>Description:</b> { this.props.description } </div>
                <div><b>Home Page:</b> <a href={this.props.homePage} target='_blank'> { this.props.homePage } </a> </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Station;