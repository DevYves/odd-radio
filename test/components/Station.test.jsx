import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Station from '../../src/client/components/Station';

describe('Station', () => {
  const props = {
    presets: {
      one: {},
      two: {},
      three: {},
      four: {},
      five: {}
    },
    key: '1',
    id: '1',
    name: 'QWER',
    description: 'radio station is what we are',
    audioFeed: 'www.google.ca',
    type: 'audio/mp3',
    homePage: 'www.example.com',
    city: 'Saskitoba',
    handleSelectedStation: jest.fn(),
    findStationExpandInfo: jest.fn(),
    stationType: 3,
    expandedName: 'QWER',
    expandedStation: false,
    clickFunction: jest.fn()
  }

  const subject = <Station { ...props }/>;

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });

  describe('fa-chevron-down onClick', () => {
    const wrapper = mount(subject);

    it('calls findStationExpandInfo', () => {

      wrapper.find('.fa-chevron-down').simulate('click');
      expect(clickFunction).toHaveBeenCalled();
    });

    it('findStationExpandInfo should set the expanded state to true', () => {
      wrapper.find('.fa-chevron-down').simulate('click');

      expect(props.findStationExpandInfo).toHaveBeenCalled();
    });
  });

  describe('when the info is expanded', () => {
    const wrapper = mount(subject);
    wrapper.setState({ expandedStation: true });

    wrapper.find('.fa-chevron-down').simulate('click');

      it('calls toggleStationInfo', () => {

        expect(wrapper.state().expandedStation).toEqual(true);
        expect(clickFunction).toHaveBeenCalled();
      });
  });
});