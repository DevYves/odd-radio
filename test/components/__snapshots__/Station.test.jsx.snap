// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station matches snapshot 1`] = `
<div
  id="QWER"
>
  <div
    className="container station-container3"
  >
    <div
      className="row station-row border"
    >
      <div
        className="four columns station-branding center"
      >
        <div
          className="non-expanded-station-name center"
          onClick={[Function]}
        >
           
           
          QWER
        </div>
      </div>
      <div
        className="four columns station-play-button center"
      >
        <div>
          <i
            className="fa fa-play fa-2x station-play-button"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="four columns station-info-button center"
      >
        <i
          aria-hidden="true"
          className="fa fa-chevron-down card-chevron fa-2x"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
